src/test/store/receive-profile.js

```diff
@@ -25,6 +46,71 @@ describe('actions/receive-profile', function () {
     });
   });
 
+  describe('retrieveProfileFromAddon', function () {
+    let geckoProfiler;
+    let clock;
+
+    beforeEach(function () {
+      clock = sinon.useFakeTimers();
```

Comment:
this replaces setTimeout and setInterval by mocks that can be controlled by `clock.tick()`. Tell me if you'd like me to add a comment about this.